tests/api/images.cpp

```diff
@@ -302,3 +302,105 @@ TEST_F(WithCommandQueue,
     EnqueueUnmapMemObject(image, data);
     Finish();
 }
+
+TEST_F(WithCommandQueue, ImageWriteOffset) {
+    // Create a 2D image.
+    const size_t IMAGE_WIDTH = 16;
+    const size_t IMAGE_HEIGHT = 16;
+
+    // This is the full image (used for initial fill and final read back).
+    size_t full_origin[3] = {0, 0, 0};
+    size_t full_region[3] = {IMAGE_WIDTH, IMAGE_HEIGHT, 1};
+
+    // We are going to write to this region.
+    size_t write_origin[3] = {IMAGE_WIDTH / 4, IMAGE_HEIGHT / 4, 0};
+    size_t write_region[3] = {IMAGE_WIDTH / 2, IMAGE_HEIGHT / 2, 1};
+
+    cl_uchar fill_value = 0xFF;
+    std::vector<cl_uchar> write_data(IMAGE_WIDTH * IMAGE_HEIGHT, 0);
```

Comment:
Suggest reducing the size of `write_data` to the size of `write_region` (and adapting the init loop nest below) to make it clear that the other data is never accessed.

Reply:
Good idea, done.